packages/neo-one-smart-contract-lib/src/ownership/withTransferableOwnership.ts

```diff
@@ -0,0 +1,77 @@
+// tslint:disable readonly-keyword readonly-array no-object-mutation strict-boolean-expressions no-unnecessary-field-initialization
+
+import { Address, constant, createEventNotifier, SmartContract } from '@neo-one/smart-contract';
+
+/**
+ * @title Transferable
+ * @dev A TransferableOwnership contract has an owner address, and provides basic authorization control
+ * functions, this simplifies the implementation of "user permissions".
+ */
+
+export function withTransferableOwnership<TBase extends Constructor<SmartContract>>(Base: TBase) {
+  abstract class TransferableOwnershipClass extends Base {
+    protected abstract mutableOwner: Address;
+    protected abstract mutableTransferTo: Address | undefined;
+
+    @constant
+    public get owner(): Address {
+      return this.mutableOwner;
+    }
+
+    private readonly notifyOwnershipTransferred = createEventNotifier<Address, Address>(
+      'contract ownership transferred',
+      'from',
+      'to',
+    );
+    private readonly notifyOwnershipTransferInitialized = createEventNotifier<Address, Address>(
+      'contract ownership transfer initiated',
```

Comment:
👍 fyi
``` [tslint] variable name must be in lowerCamelCase, PascalCase or UPPER_CASE [variable-name]
 ```

Reply:
Maybe, but then this guy commented below suggesting... 
![image](https://user-images.githubusercontent.com/39564353/52579536-31e9bb80-2ddb-11e9-999e-e5b49efdcd33.png)
for the same line.  I'll go with the camelCase because

1.) it matches the courses 
2.) the below suggestions match it.
3.) the id complains about the words_with_underscores, wanting camelCase.

Reply:
## using ` snake_case ` for notifications
even though that doesn't match the notification functions in the Token.ts